### ISSUE TITLE
Title only callout in revealjs causes error

### ISSUE THREAD

**Issue by CMFell** [quarto-dev/quarto-cli#9821]
### Bug description

If you create a callout-note with just a title in revealjs format it does not render and gives a lua filter error.
This will correctly create a callout with just a title in html format.

### Steps to reproduce

````
---
format: revealjs
---

The following call out doesn't cause a crash

::: {.callout-note}

## Test

Some text

:::
````

````
---
format: revealjs
---

The following call out causes a crash

::: {.callout-note}

## Test

:::
````

### Expected behavior

Create a callout with just a title as happens in html format. 

Or if not possible give a warning instead of an error

### Actual behavior

Error running filter C:/Users/cmf21/AppData/Local/Programs/Quarto/share/filters/main.lua:
Block, list of Blocks, or compatible element expected, got nil
	while retrieving function argument content
	while retrieving arguments for function Div
stack traceback:
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:20877: in field 'render'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1233: in local 'filter_fn'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:635: in function <...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:625>
	(...tail calls...)
	[C]: in ?
	[C]: in method 'walk'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:557: in function <...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:546>
	(...tail calls...)
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1334: in local 'callback'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1352: in upvalue 'run_emulated_filter_chain'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1388: in function <...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1385>
stack traceback:
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:557: in function <...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:546>
	(...tail calls...)
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1334: in local 'callback'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1352: in upvalue 'run_emulated_filter_chain'
	...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1388: in function <...f21/AppData/Local/Programs/Quarto/share/filters/main.lua:1385>

### Your environment

-IDE RStudio 2024.04.0+735 - windows 11

### Quarto check output

Quarto 1.5.39
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.39
      Path: C:\Users\cmf21\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\cmf21\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.5 (Conda)
      Path: C:/Users/cmf21/AppData/Local/anaconda3/envs/id5059/python.exe
      Jupyter: 5.5.0
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.3.2
      Path: C:/PROGRA~1/R/R-43~1.2
      LibPaths:
        - C:/Users/cmf21/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.2/library
      knitr: 1.46
      rmarkdown: 2.27

[>] Checking Knitr engine render......OK

**Comment by cderv** [comment]
Interesting thanks ! 

This is related to 

* #9763  (#9772)

I believe callout with empty body are not well supported accross format

**Comment by debdagybra** [comment]
It works fine on my computer.

```
---
format: revealjs
---

::: {.callout-note}
## Test callout title only
:::

::: {.callout-note appearance="default"}
## Test callout title only
:::

::: {.callout-note appearance="minimal"}
## Test callout title only
:::

::: {.callout-note appearance="simple"}
## Test callout title only
:::
```

![image](https://github.com/quarto-dev/quarto-cli/assets/10846318/0b0124b0-e087-4edd-a66a-88e5f94d2760)

Windows 11 and IDE RStudio/2024.04.1+748 

## Quarto check output

Quarto 1.5.26
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.26
      Path: C:\Users\damie\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: v2024.05
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\damie\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.4.0
      Path: C:/PROGRA~1/R/R-44~1.0
      LibPaths:
        - C:/Users/damie/AppData/Local/R/win-library/4.4
        - C:/Program Files/R/R-4.4.0/library
      knitr: 1.47
      rmarkdown: 2.27

[>] Checking Knitr engine render......OK

**Comment by mcanouil** [comment]
@debdagybra you are using an outdated pre-release version.

**Comment by debdagybra** [comment]
Sorry I missed that ! Indeed, it doesn't work with 1.5.39

**Comment by cderv** [comment]
So interesting to see that it was working. Must be recent change !

**Comment by cderv** [comment]
This is related to more global issue 
* #9832 

We'll probably resolved as once

**Comment by psobolewskiPhD** [comment]
I just ran into this when updated from `quarto 1.4.553 -> 1.5.56`
Took me a while to debug, basically bisecting using `<!--` and `-->`
Is there a better way to get info what line in the .qmd is causing render errors like this one?

**Comment by cscheid** [comment]
> Is there a better way to get info what line in the .qmd is causing render errors like this one?

Unfortunately not - this is a pretty serious Quarto limitation right now, but we can't really do much because the line information is not preserved by Pandoc (a tool we use internally in Quarto). As a result, we don't actually have access to that information when the error happens.

**Comment by cscheid** [comment]
Sorry I missed this issue earlier. The cause of this bug is the update to Pandoc 3.2, which fixed a bug in Lua that causes accesses to Div fields to mutate their content from `nil` to `Plain []`. After this bug is fixed, a number of our elements become (correctly) `nil`, but we now need to check for them. We were missing this check in the revealjs callout renderer.

**Comment by cderv** [comment]
This has been fix and backported to 1.5, so it will be available in next 1.5 patch release.
